app/services/package.scala

```diff
@@ -14,31 +14,38 @@
  * limitations under the License.
  */
 
-import services.shared._
-import models.ocelot._
-import models.errors.Error
-import models.ocelot.errors._
-import models.RequestOutcome
-import models.ocelot.Process
+import core.services._
+import core.models.ocelot._
+import core.models.errors.Error
+import core.models.ocelot.errors._
+import core.models.RequestOutcome
+import core.models.ocelot.Process
 import play.api.libs.json._
 import config.AppConfig
 
 package object services {
-  def guidancePages(pageBuilder: PageBuilder, securedProcessBuilder: SecuredProcessBuilder, jsObject: JsObject)
-                   (implicit c: AppConfig): RequestOutcome[(Process, Seq[Page], JsObject)] =
-    jsObject.validate[Process].fold(
-      errs => Left(Error(GuidanceError.fromJsonValidationErrors(errs))),
-      p => {
-        val process = fakeWelshText(p.passPhrase.fold(p)(_ => securedProcessBuilder.secure(p)))
-        pageBuilder.pagesWithValidation(process, process.startPageId).fold(
+  def guidancePages(pageBuilder: PageBuilder, jsObject: JsObject)
+                   (implicit c: AppConfig, spd: SecuredProcessBuilder): RequestOutcome[(Process, Seq[Page], JsObject)] =
+    jsObject.validate[Process].fold(errs => Left(Error(GuidanceError.fromJsonValidationErrors(errs))),
+      incomingProcess => {
+        // Transform process if fake welsh required and/or secured process is indicated
+        val (p, js) = fakeWelshTextIfRequired _ tupled securedProcessIfRequired(incomingProcess, jsObject)
+        pageBuilder.pagesWithValidation(p, p.startPageId).fold(
           errs => Left(Error(errs)),
-          pages => Right((process, pages, Json.toJsObject(process)))
+          pages => Right((p, pages, js))
         )
       }
     )
 
-  def fakeWelshText(process: Process)(implicit c: AppConfig): Process =
-    if (!process.passPhrase.isEmpty || c.fakeWelshInUnauthenticatedGuidance)
-      process.copy(phrases = process.phrases.map(p => if (p.welsh.trim.isEmpty) Phrase(p.english, s"Welsh, ${p.english}") else p))
-    else process
+  private [services] def fakeWelshTextIfRequired(process: Process, jsObject: JsObject)(implicit c: AppConfig): (Process, JsObject) =
+    if (!process.passPhrase.isEmpty || c.fakeWelshInUnauthenticatedGuidance) {
```

Comment:
According to IntelliJ you can use process.passPhrase.isDefined here.

Reply:
yes